src/source.cpp

```diff
@@ -38,6 +38,10 @@ namespace openmc {
 
 namespace model {
 
+typedef Particle::Bank (*sample_t)(uint64_t  &seed);
+sample_t sample_source;
+void* source_library;
```

Comment:
Maybe the following names would make things a little more obvious
```suggestion
sample_t custom_source_function;
void* custom_source_library;
```